metaflow/environment.py

```diff
@@ -76,21 +76,15 @@ def get_client_info(cls, flow_name, metadata):
         """
         return "Local environment"
 
-    def get_package_commands(self, code_package_url):
+    def get_package_commands(self, code_package_url,datastore):
         cmds = ["set -e",
                 "echo \'Setting up task environment.\'",
                 "%s -m pip install awscli click requests boto3 \
```

Comment:
It may make sense to move the awscli and boto3 requirements into the package_download_commands as well since they are not related to the rest here (requests and click are MF requirements independent of AWS/S3)

Reply:
Yes. That makes sense. Do you think abstractions should also apply the Decorators like `BatchDecorator`?